test/EndToEndTests/Microsoft.OData.E2E.Portable/AsynchronousTests/AsynchronousQueryTests.cs

```diff
@@ -1197,6 +1197,38 @@ public void Linq_Where_GreaterThan()
             this.WaitForTestToComplete();
         }
 
+        /// <summary>
+        /// When DataServiceContext.IdPredicateGeneratesFilterQueryOption=true, An expression with an Id as a predicate should create an Uri with a $filter in query options
+        /// </summary>
+        [Fact]
+        public void Linq_Where_Generates_Filter_When_IdPredicateGeneratesFilterQueryOption_Is_True()
+        {
+            var context = this.CreateWrappedContext<DefaultContainer>().Context;
+            context.IdPredicateGeneratesFilterQueryOption = true;
+            var query = (from c in context.Customer
+                         where c.CustomerId == -10
+                         select new Customer { Name = c.Name, CustomerId = c.CustomerId }) as DataServiceQuery<Customer>;
+            var uri = query.ToString();
+            Assert.Contains("$filter=CustomerId eq -10", uri);
+            Assert.Equal("$filter=CustomerId eq -10", uri);
+        }
+
+        /// <summary>
+        /// When DataServiceContext.IdPredicateGeneratesFilterQueryOption=false, An expression with an Id as a predicate should create a ByKey Uri
+        /// </summary>
+        [Fact]
+        public void Linq_Where_Generates_ByKey_When_IdPredicateGeneratesFilterQueryOption_Is_False()
+        {
+            var context = this.CreateWrappedContext<DefaultContainer>().Context;
+            context.IdPredicateGeneratesFilterQueryOption = false;
+            var query = (from c in context.Customer
+                         where c.CustomerId == -10
+                         select new Customer { Name = c.Name, CustomerId = c.CustomerId }) as DataServiceQuery<Customer>;
+            var uri = query.ToString();
+            Assert.Contains("Customer(-10)", uri);
+            Assert.Equal("Customer(-10)", uri);

```

Comment:
`Assert.Equal` should be adequate. Same for previous test

Reply:
Fixed